### PR TITLE
refactor(*): move constants into /libs/constants and env vars into /libs/env

### PR DESCRIPTION
**Pull request by caugner**
## Summary

Decouples yari folders by moving constants and environment variables into `/libs/*`. All constant values go into `/libs/constants`, and all environment variables go into `/libs/env` (**new**).

### Background

We use several constants (e.g. `MDN_PLUS_TITLE`) and environment variables (e.g. `CONTENT_ROOT`) across yari.

### Problems

1. We determine, export and import these "all over the place", which unnecessarily increases coupling of unrelated components/folders.
2. We use the term "constants" for both _constant values_ and _environment variables_.

### Solution

1. Move all constant values into `/libs/constants`.
2. Move all environment variables into `/libs/env` (except for `REACT_APP_*` vars, which are special).
3. In `/client`, distinguish `./constants.ts` and `./env.ts`.

**Note**: I had to disarm the check that would previously fail if `CONTENT_ROOT` or `CONTENT_TRANSLATED_ROOT` were set to an invalid path, so that parts of our code can use `/libs/env` without setting these environment variables.

---

## How did you test this change?

Ran `yarn dev` without problems, and tests are passing.

### CODE REVIEW

File: libs/env/index.js [hunk 1]
```diff
@@ -1,13 +1,19 @@
+const fs = require("fs");
```
Comment (caugner) [inline libs/env/index.js:1]: @fiji-flo Does it make sense to distinguish between ENV variables (`lib/env`) and constants (`lib/constants`), or should I just merge them into one?

Comment (caugner) [reply libs/env/index.js:1]: (I think it makes sense because `lib/constants` doesn't depend on any dotenv/fs/path import.)

**Comment by caugner** [comment]
TODO Wait for #6385, then apply same changes to `client/`.

File: libs/env/index.js [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+// -----
+
+const BUILD_OUT_ROOT =
+  process.env.BUILD_OUT_ROOT || path.join(ROOT, "client", "build");
+
+const FLAW_LEVELS = Object.freeze({
+  ERROR: "error",
+  IGNORE: "ignore",
+  WARN: "warn",
+});
```
Comment (caugner) [inline libs/env/index.js:24]: Move to `libs/constants`.

File: libs/env/index.js [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+  "bad_bcd_queries",
+  "bad_bcd_links",
+  "images",
+  "image_widths",
+  "bad_pre_tags",
+  "sectioning",
+  "heading_links",
+  "translation_differences",
+  "unsafe_html",
+]);
```
Comment (caugner) [inline libs/env/index.js:46]: ditto

File: libs/env/index.js (hunk 2 above)
Comment (caugner) [reply libs/env/index.js]: done

File: libs/env/index.js [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+// need to constantly worry about CONTENT_TRANSLATED_ROOT potentially being
+// null.
+const ROOTS = [CONTENT_ROOT];
+if (CONTENT_TRANSLATED_ROOT) {
+  ROOTS.push(CONTENT_TRANSLATED_ROOT);
+  REPOSITORY_URLS[CONTENT_TRANSLATED_ROOT] = "mdn/translated-content";
+}
+
+const HTML_FILENAME = "index.html";
+const MARKDOWN_FILENAME = "index.md";
```
[the remark this answers is not in the document]
Comment (caugner) [reply libs/env/index.js]: done

File: libs/env/index.js (hunk 3 above)
Comment (caugner) [reply libs/env/index.js]: done

**Review by fiji-flo — APPROVED** [review]
Looks great. Can you do a stage deployment after merging?

**Pull request merged**